src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * External dependencies
+ */
+import { OpentableIcon as icon } from '@godaddy-wordpress/coblocks-icons';
+import { useState, useEffect } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import Controls from './controls';
+import OpenTable from './opentable';
+import InspectorControls from './inspector';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+import {
+	Placeholder,
+	Button,
+	withNotices,
+	Icon,
+	FormTokenField,
+} from '@wordpress/components';
+
+const Edit = ( props ) => {
+	// const onSelectImages = ( images ) => {
+	// 	props.setAttributes( { images: images.map( ( image ) => pickRelevantMediaFiles( image ) ) } );
+	// };
+
+	// const onDropImages = ( images ) => {
+	// 	const imagesNormalized = images.map( ( image ) => pickRelevantMediaFiles( image ) );
+	// 	const currentImages = props.attributes.images || [];
+	// 	props.setAttributes( { images: currentImages.concat( imagesNormalized )	} );
+	// };
+
+	// const { className, noticeOperations, attributes, noticeUI, isSelected } = props;
+
+	const [ ridField, setRidField ] = useState( [] );
+
+	const [ queryResults, setQueryResults ] = useState( [] );
+	const { className, attributes } = props;
+
+	//searches the opentable reservation network for restaurants with the given name or ID
+	const searchRestaurants = ( token ) => {
+		fetch(
+			'https://www.opentable.com/widget/reservation/restaurant-search?pageSize=15' +
+				'&query=' +
+				encodeURIComponent( token )
+		)
+			.then( ( response ) => response.json() )
+			.then( ( json ) => {
+				setQueryResults( [] );
+				for ( const item in json.items ) {
+					const name = decodeURIComponent( json.items[ item ].name ) + ' (' + json.items[ item ].rid + ')';
+					setQueryResults( ( oldQueryResults ) => [ ...oldQueryResults, name ] );
+				}
+			} );
+	};
+	useEffect( () => {
+		setRidField( attributes.restaurantIDs );
+	}, [] );
```

Comment:
This `useEffect` call would fire on every render thus `setRidField( attributes.restaurantIDs );` would fire on every render. This could be solved by adding `attributes.restaurantIDs` to the dependency array on line 62 and by adding 
```
if ( prevIDs !== attributes.restaurantIDs ) {

}
```
conditional on the setter. 

You can access previous props like this.
https://github.com/godaddy-wordpress/coblocks/blob/aacd25ceb1cf712bbe1f852f197922ca6246de85/src/blocks/gallery-collage/edit.js#L32-L33

Reply:
Question on this - since the useEffect array parameter is an empty array rather than being not provided at all, wouldn't the above code only fire on the first render?

Reply:
Yes, it should act as a `componentDidMount()`

Reply:
My mistake! 👍